pkg/controller/dataimportcron-controller.go

```diff
@@ -420,12 +457,35 @@ func (r *DataImportCronReconciler) getImportState(ctx context.Context, cron *cdi
 	return dv, pvc, nil
 }
 
-func (r *DataImportCronReconciler) updatePvc(ctx context.Context, cron *cdiv1.DataImportCron, pvc *corev1.PersistentVolumeClaim) error {
-	pvcCopy := pvc.DeepCopy()
-	cc.AddAnnotation(pvc, AnnLastUseTime, time.Now().UTC().Format(time.RFC3339Nano))
-	r.setDataImportCronResourceLabels(cron, pvc)
-	if !reflect.DeepEqual(pvc, pvcCopy) {
-		if err := r.client.Update(ctx, pvc); err != nil {
+// Returns the current import DV if exists, and the last imported PVC
+func (r *DataImportCronReconciler) getSnapshot(ctx context.Context, cron *cdiv1.DataImportCron, format cdiv1.DataImportCronSourceFormat) (*snapshotv1.VolumeSnapshot, error) {
+	if format != cdiv1.DataImportCronSourceFormatSnapshot {
+		return nil, nil
+	}
+
+	imports := cron.Status.CurrentImports
+	if len(imports) == 0 {
+		return nil, nil
+	}
+
+	snapName := imports[0].DataVolumeName
+	snapshot := &snapshotv1.VolumeSnapshot{}
+	if err := r.client.Get(ctx, types.NamespacedName{Namespace: cron.Namespace, Name: snapName}, snapshot); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return nil, err
+		}
+		snapshot = nil
+	}
+
+	return snapshot, nil
+}
+
+func (r *DataImportCronReconciler) updateSourceObject(ctx context.Context, cron *cdiv1.DataImportCron, obj client.Object) error {
```

Comment:
Not sure the word `Object` adds anything. All update functions update an object :)

Reply:
removed